### PR TITLE
ch4/ofi: Make pack buffer pool independent of active messages

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Pack buffers from this pool are utilized in RMA code. Rename the pool
and associated CVARs to be independent of actives messages. Move
init/finalize outside of AM conditionals.

## Expected Impact

none

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_types.h [hunk 1]
```diff
@@ -384,6 +383,9 @@ typedef struct {
     /* Queue (utlist) to store early-arrival active messages */
     MPIDI_OFI_am_unordered_msg_t *am_unordered_msgs;
 
+    /* Pack buffers for various communication */
+    MPIDU_genq_private_pool_t am_pack_buf_pool;
```
Comment (yfguo) [inline src/mpid/ch4/netmod/ofi/ofi_types.h:387]: Should be `pack_buf_pool` here.

Comment (yfguo) [reply src/mpid/ch4/netmod/ofi/ofi_types.h:387]: Also, there are other places that use this pack buffer (ofi_am_impl.h/c, etc.)

**Comment by raffenet** [comment]
test:mpich/ch4/ofi

**Comment by raffenet** [comment]
test:mpich/ch4/ofi

**Pull request merged**